src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -0,0 +1,624 @@
+import 'angular/core/services/translation.service';
+import {QueryType} from "../../../../../models/ontotext-yasgui/query-type";
+import {BeforeUpdateQueryResult, BeforeUpdateQueryResultStatus} from "../../../../../models/ontotext-yasgui/before-update-query-result";
+import {queryPayloadFromEvent, savedQueriesResponseMapper} from "../../../rest/mappers/saved-query-mapper";
+import {isFunction, merge} from "lodash";
+import {downloadAsFile, toYasguiOutputModel} from "../../../utils/yasgui-utils";
+import {EventDataType} from "../../../../../models/ontotext-yasgui/event-data-type";
+import {QueryMode} from "../../../../../models/ontotext-yasgui/query-mode";
+import {YasrPluginName} from "../../../../../models/ontotext-yasgui/yasr-plugin-name";
+import {isEqual} from "lodash/lang";
+
+const modules = ['graphdb.framework.core.services.translation-service'];
+angular
+    .module('graphdb.framework.core.directives.yasgui-component', modules)
+    .directive('yasguiComponent', yasguiComponentDirective);
+
+yasguiComponentDirective.$inject = [
+    '$q',
+    '$repositories',
+    '$translate',
+    '$location',
+    '$uibModal',
+    '$languageService',
+    '$jwtAuth',
+    'toastr',
+    'TranslationService',
+    'AutocompleteRestService',
+    'RDF4JRepositoriesRestService',
+    'MonitoringRestService',
+    'ConnectorsRestService',
+    'SparqlRestService',
+    'ShareQueryLinkService'
+];
+
+function yasguiComponentDirective(
+    $q,
+    $repositories,
+    $translate,
+    $location,
+    $uibModal,
+    $languageService,
+    $jwtAuth,
+    toastr,
+    TranslationService,
+    AutocompleteRestService,
+    RDF4JRepositoriesRestService,
+    MonitoringRestService,
+    ConnectorsRestService,
+    SparqlRestService,
+    ShareQueryLinkService
+) {
+
+    const HEADERS = {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'Accept': 'application/sparql-results+json',
+        'X-GraphDB-Local-Consistency': 'updating'
+    };
+
+    const DEFAULT_CONFIG = {
+        showEditorTabs: true,
+        showToolbar: true,
+        componentId: 'yasgui-component',
+        headers: () => {
+            return HEADERS;
+        },
+        pageSize: 1000,
+        maxPersistentResponseSize: 500000,
+        showResultTabs: true
+    };
+
+    return {
+        restrict: 'AE',
```

Comment:
I believe that we can restrict the selector to just `E`lement. We don't need the `A`ttribute one.

Reply:
Done